### PR TITLE
BLD: Add meson build scripts

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

In preparation for the rest of the meson build system support.

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
Lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks, just some questions to follow up

File: scripts/generate_version.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import argparse
+import os
+
+import versioneer
+
+
+def write_version_info(path):
+    if os.environ.get("MESON_DIST_ROOT"):
+        # raise ValueError("dist root is", os.environ.get("MESON_DIST_ROOT"))
```
Comment (mroeschke) [inline scripts/generate_version.py:9]: If you could remove in a follow up that would be great

File: scripts/generate_version.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import argparse
+import os
+
+import versioneer
```
Comment (mroeschke) [inline scripts/generate_version.py:4]: So this won't be our vendored versioneer?

Comment (lithomas1) [reply scripts/generate_version.py:4]: Oops. I forgot to update this when I moved it into scripts, will fix it in my next PR.

Unfortunately, I don't think we are able to use non-vendored versioneer.

**Pull request merged**